opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -543,16 +555,25 @@ namespace Opm {
                         sg = 1.0 - sw - so;
                         double& rv = reservoir_state.rv()[cell_idx];
                         // use gas pressure?
-                        double rvSat = FluidSystem::gasPvt().saturatedOilVaporizationFactor(0, reservoir_state.temperature()[cell_idx], reservoir_state.pressure()[cell_idx]);
+                        double rvSat = FluidSystem::gasPvt().saturatedOilVaporizationFactor(fs.pvtRegionIndex(), reservoir_state.temperature()[cell_idx], reservoir_state.pressure()[cell_idx]);
                         rv = rvSat*(1-epsilon);
                     }
                     break;
                 }
                 case HydroCarbonState::OilOnly: {
-                    double& rs = reservoir_state.gasoilratio()[cell_idx];
-                    double& sg = reservoir_state.saturation()[cell_idx*np + pu.phase_pos[ Gas ]];
-                    // TODO:: not hardcode pvtRegion = 0
-                    double rsSat = FluidSystem::oilPvt().saturatedGasDissolutionFactor(0, reservoir_state.temperature()[cell_idx], reservoir_state.pressure()[cell_idx]);
+                    if (sw > (1.0 - epsilon)) {
+                        // water only change to Sg
```

Comment:
I agree. I just noticed because it recently caused some issues in the threshold pressure code...